server/server.go

```diff
@@ -24,18 +23,59 @@ import (
 const Version = "2.0.1b"
 
 type server struct {
-	client       *etcd.Client
+	getter Getter
+	config *Config
+
+	group        *sync.WaitGroup
 	dnsUDPclient *dns.Client // used for forwarding queries
 	dnsTCPclient *dns.Client // used for forwarding queries
-	config       *Config
-	group        *sync.WaitGroup
 	scache       *cache.Cache
 	rcache       *cache.Cache
 }
 
+type Getter interface {
+	Records(name string, exact bool) ([]msg.Service, error)
```

Comment:
Agree, I was only using your example :)

Backend or RecordRetriever or RecordLoader or RecordStore all work for me.


> On Mar 8, 2015, at 10:42 AM, Miek Gieben <notifications@github.com> wrote:
> 
> In server/server.go:
> 
> >  	scache       *cache.Cache
> >  	rcache       *cache.Cache
> >  }
> >  
> > +type Getter interface {
> > +	Records(name string, exact bool) ([]msg.Service, error)
> Getter sounds so... java-ish, maybe Backend or Backender?
> 
> —
> Reply to this email directly or view it on GitHub.
>